tests/test_csv_exporters/test_application_round_results_export.py

```diff
@@ -0,0 +1,218 @@
+        reservations_end_date=local_date_string(section.reservations_end_date),
+        applied_reservations_per_week=str(section.applied_reservations_per_week),
+        reservation_min_duration=local_timedelta_string(section.reservation_min_duration),
+        reservation_max_duration=local_timedelta_string(section.reservation_max_duration),
+        reservation_unit_name="foo",
+        unit_name="fizz",
+        day_of_the_week=allocation.day_of_the_week,
+        begin_time="12:00",
+        end_time="14:00",
+        price="",
+    )
+
+
+def test_application_round_results_export__no_suitable_time_ranges(graphql):
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    ApplicationFactory.create_in_status_in_allocation(
+        application_round=application_round,
+    )
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    assert exporter.export() is None
+
+
+def test_application_round_results_export__application_status_is_expired(graphql):
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    ApplicationFactory.create_in_status_expired(
+        application_round=application_round,
+        application_sections__suitable_time_ranges__day_of_the_week=Weekday.TUESDAY,
+    )
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    assert exporter.export() is None
+
+
+def test_application_round_results_export__application_status_is_draft(graphql):
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    ApplicationFactory.create_in_status_draft(
+        application_round=application_round,
+        application_sections__suitable_time_ranges__day_of_the_week=Weekday.TUESDAY,
+    )
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    assert exporter.export() is None
+
+
+def test_application_round_results_export__section_has_no_allocated_time_slots(graphql):
+    """Section which has no allocated time slots should still be exported."""
+    application_round = ApplicationRoundFactory.create_in_status_handled()
+    application = ApplicationFactory.create_in_status_in_allocation(application_round=application_round)
+    section = ApplicationSectionFactory.create_in_status_handled(
+        application=application,
+        suitable_time_ranges__day_of_the_week=Weekday.WEDNESDAY,
+        reservation_unit_options__reservation_unit__name="foo",
+    )
+
+    exporter = ApplicationRoundResultCSVExporter(application_round_id=application_round)
+    with mock.patch(CSV_WRITER_MOCK_PATH) as mock_writer:
+        exporter.export()
+    writes = get_writes(mock_writer)
+    assert len(writes) == 2
+
+    section_row: ApplicationSectionExportRow = writes[1]  # type: ignore
+    assert section_row.section_id == str(section.id)
+    assert section_row.reservation_unit_name == "foo"
```

Comment:
Since there would be a single row in the export when a section has no allocations at all, regardless of how many (unallocated) res.unit options it has, the `reservation_unit_name column` too would be empty. Cf. comment in the exported class: https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1157/files#r1570069055